### PR TITLE
#18316 Removing duplicates from query

### PR DESCRIPTION
**Pull request by nollymar**
This PR fixes several issues: 
- `-baseType:3` was being appended twice in lucene queries
- `-contentType:forms` was being appended even when not needed, ie.: when the lucene query contained a `+contentType: blog`
- Filter by `-contentType:host` was removed
- Logic to filter content types by License Level was moved to the ESContentFactoryImpl class
- when "All" content types were selected in the Content Search portlet, each content type was being appended in the lucene query instead of using `*`
- when filtering by title, the query builder was using multiple boost factors `^` for the same field, ie.: `title:'engl'^15 title:engl^5`

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java [hunk 1]
```diff
@@ -892,10 +895,12 @@ else if( fieldbcontentname.startsWith("date") ){
...
-							            for(String x : fieldValueStr.split("[,|\\s+]")) {
-							                luceneQuery.append(" title:" + x + "^5 ");
-							                
-							            }
+                                        final String[] titleSplit = fieldValueStr.split("[,|\\s+]");
+                                        if (titleSplit.length > 1) {
+                                            for (final String term : titleSplit) {
+                                                luceneQuery.append(" title:" + term + "^5 ");
+                                            }
+                                        }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java:903]: all the 'searchContentletsByUser' method code should go into a API

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java [hunk 2]
```diff
@@ -892,10 +895,12 @@ else if( fieldbcontentname.startsWith("date") ){
...
 							            luceneQuery.append(" title_dotraw:*" + fieldValueStr + "*^5 ");
 							        }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java:906]: any test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java:884]: I agree but the logic on this Ajax is heavy to refactor

**Comment by nollymar** [comment]
@wezell can you please take a look at these changes?

**Review by freddyucv — APPROVED** [review]

**Pull request merged**